### PR TITLE
IDEMPIERE-5204 Cache the list/array of mreflist for AD_Reference_ID

### PR DESCRIPTION
**Pull request by igorpojzl**
Jira Ticket: https://idempiere.atlassian.net/browse/IDEMPIERE-5204
Mattermost Topic: https://mattermost.idempiere.org/idempiere/pl/wh3fooer7b8ityc43edopuxmta

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/model/MRefList.java [hunk 1]
```diff
@@ -54,12 +59,44 @@
 	 *	@param trxName transaction
 	 *	@return List or null
 	 */
-	public static MRefList get (Properties ctx, int AD_Reference_ID, String Value, String trxName)
+	public static synchronized MRefList get (Properties ctx, int AD_Reference_ID, String Value, String trxName)
 	{
-		return new Query(ctx, Table_Name, "AD_Reference_ID=? AND Value=?", trxName)
+		String key = AD_Reference_ID + "|" + Value;
```
Comment (CarlosRuiz-globalqss) [inline org.adempiere.base/src/org/compiere/model/MRefList.java:64]: See comment
https://idempiere.atlassian.net/browse/IDEMPIERE-5204?focusedCommentId=48051

Comment (igorpojzl) [reply org.adempiere.base/src/org/compiere/model/MRefList.java]: Commited changes.

**Review by CarlosRuiz-globalqss — APPROVED** [review]
Hi @hengsin - for me it looks good, but you know better the details of the Cache implementation, can you please check and merge if everything is fine?

Regards,

Carlos Ruiz

**Review by hengsin — APPROVED** [review]
looks good

**Pull request merged**